docs/source/dataframe-design.rst

```diff
@@ -0,0 +1,109 @@
+Internal Design
+===============
+
+Dask dataframes coordinate many Pandas arrays arranged along an index. We
+define a ``dask.dataframe`` object with the following components:
+
+- A dask graph with a special set of keys designating partitions, such as
+  ``('x', 0), ('x', 1), ...``.
+- A name to identify which keys in the dask graph refer to this dataframe, such
+  as ``'x'``.
+- An empty pandas object containing appropriate metadata (e.g.  column names,
+  dtypes, etc...).
+- A sequence of partition boundaries along the index.
+
+Metadata
+--------
+
+Many dataframe operations rely on knowing the name and dtype of columns. To
+keep track of this information, all ``dask.dataframe`` objects have a ``_meta``
+attribute which contains an empty object with the same dtypes and names. For
```

Comment:
Should perhaps specify "an empty *pandas* object" or other term to designate that this is a normal non-dask thing